launch/src/main/scala/xsbt/boot/ServerApplication.scala

```
@@ -0,0 +1,166 @@
+  def locate(currentDirectory: File, config: LaunchConfiguration): URI = 
+    config.serverConfig match {
+      case None => sys.error("No server lock file configured.  Cannot locate server.")
+      case Some(sc) => locked(makeLockFile(sc.lockFile)) {
+        readProperties(sc.lockFile) match {
+          case Some(uri) if isReachable(uri) => uri
+          case _ => 
+            val uri = ServerLauncher.startServer(currentDirectory, config)
+            writeProperties(sc.lockFile, uri)
+            uri
+        }
+      }
+    }
+  
+  private val SERVER_URI_PROPERTY = "server.uri"
+  def readProperties(f: File): Option[java.net.URI] = {
+    try {
+      val props = new java.util.Properties
+      val in = new java.io.FileInputStream(f)
+      try {
+        props.load(in)
+      } finally in.close()
+      props.getProperty(SERVER_URI_PROPERTY) match {
+        case null => None
+        case uri  => Some(new java.net.URI(uri))
+      }
+    } catch {
+      case e: IOException => None
+    }
+  }
+  def writeProperties(f: File, uri: URI): Unit = {
+    val props = new java.util.Properties
+    props.setProperty(SERVER_URI_PROPERTY, uri.toASCIIString)
+    val output = new java.io.FileOutputStream(f)
+    // TODO - Better date format.
+    try props.store(output, s"Server Startup at ${new java.util.Date()}")
+    finally output.close()
+  }
+  
+  def isReachable(uri: java.net.URI): Boolean = 
+    try {
+      // TODO - For now we assume if we can connect, it means
+      // that the server is working...
+      val socket = new java.net.Socket(uri.getHost, uri.getPort)
+      try socket.isConnected
+      finally socket.close()
+    } catch {
+      case e: IOException => false
+    }
+}
+object ServerLauncher {
+  import ServerApplication.SERVER_SYNCH_TEXT
+  def startServer(currentDirectory: File, config: LaunchConfiguration): URI = {
+    // TODO - better error message on failure.  We shouldn't make it here if this is not true...
```

I guess the point is there's some fix for the todo that's shorter than the todo.
maybe getOrElse(throw "please report bug") or just let the get throw